include/swift/Basic/SourceLoc.h

```diff
@@ -188,10 +188,9 @@ class CharSourceRange {
 
   /// Returns true if the given source location is contained in the range.
   bool contains(SourceLoc loc) const {
-    auto less = std::less<const char *>();
     auto less_equal = std::less_equal<const char *>();
     return less_equal(getStart().Value.getPointer(), loc.Value.getPointer()) &&
-           less(loc.Value.getPointer(), getEnd().Value.getPointer());
+           less_equal(loc.Value.getPointer(), getEnd().Value.getPointer());
```

Comment:
This is not correct. `CharSourceRange` is a half open range (`SourceRange` is a closed range). so `getEnd()` is not part of the range.

Reply:
Indeed, as the tests tell me. The underlying problem here was that EOF isn't considered within the range for `getSourceFileContainingLocation`, but I've just updated that to instead check for `getEnd()` as well.